soseki/reader/modeling.py

```diff
@@ -20,27 +21,19 @@
 
 
 class ReaderModel(nn.Module):
-    def __init__(
-        self,
-        base_model_name: str,
-        pooling_index: int = 0,
-    ) -> None:
+    def __init__(self, base_model_name: str, pooling_index: int = 0) -> None:
         super().__init__()
         self.encoder = AutoModel.from_pretrained(base_model_name, add_pooling_layer=False, return_dict=False)
-        # For LUKE / mLUKE models which take large amounts of memory for entity embeddings
-        if hasattr(self.encoder, "entity_embeddings"):
-            del self.encoder.entity_embeddings
+        self.pooling_index = pooling_index
 
         self.qa_outputs = nn.Linear(self.encoder.config.hidden_size, 2)
         self.qa_classifier = nn.Linear(self.encoder.config.hidden_size, 1)
 
-        self._pooling_index = pooling_index
-
-    def forward(self, input_ids: Tensor, attention_mask: Tensor, token_type_ids: Tensor) -> Tuple[Tensor]:
-        encoder_output = self.encoder(input_ids, attention_mask=attention_mask, token_type_ids=token_type_ids)
+    def forward(self, input_tensors: Dict[str, Tensor]) -> Tuple[Tensor, Tensor, Tensor]:
+        encoder_output = self.encoder(**input_tensors)
```

Comment:
The previous model had only accepted the outputs of BERT. Now it accepts tensors that (basically) any Hugging Face model outputs.